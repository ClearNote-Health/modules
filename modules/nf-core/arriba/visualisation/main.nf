// TODO nf-core: If in doubt look at other nf-core/modules to see how we are doing things! :)
//               https://github.com/nf-core/modules/tree/master/modules/nf-core/
//               You can also ask for help via your pull request or on the #modules channel on the nf-core Slack workspace:
//               https://nf-co.re/join
// TODO nf-core: A module file SHOULD only define input and output files as command-line parameters.
//               All other parameters MUST be provided using the "task.ext" directive, see here:
//               https://www.nextflow.io/docs/latest/process.html#ext
//               where "task.ext" is a string.
//               Any parameters that need to be evaluated in the context of a particular sample
//               e.g. single-end/paired-end data MUST also be defined and evaluated appropriately.
// TODO nf-core: Software that can be piped together SHOULD be added to separate module files
//               unless there is a run-time, storage advantage in implementing in this way
//               e.g. it's ok to have a single module for bwa to output BAM instead of SAM:
//                 bwa mem | samtools view -B -T ref.fasta
// TODO nf-core: Optional inputs are not currently supported by Nextflow. However, using an empty
//               list (`[]`) instead of a file can be used to work around this issue

process ARRIBA_VISUALISATION {
    tag "$meta.id"
    label 'process_medium'

    // TODO nf-core: List required Conda package(s).
    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
    conda "bioconda::arriba=2.4.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/arriba:2.4.0--h6b7c446_1':
        'biocontainers/arriba:2.4.0--h6b7c446_1' }"

    input:
    tuple val(meta), path(bam), path(bai), path(fusions)
    tuple val(meta2), path(gtf)
    tuple val(meta3), path(protein_domains)
    tuple val(meta4), path(cytobands)

    output:
    tuple val(meta), path("*.pdf"), emit: pdf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def cytobands = cytobands ? " --cytobands=$cytobands" : ""
    def prefix = task.ext.prefix ?: "${meta.id}"
    def protein_domains = protein_domains ? "--proteinDomains=$protein_domains" : ""
    """
    draw_fusions.R \\
        --fusions=$fusions \\
        --alignments=$bam \\
        --output=${prefix}.pdf \\
        --annotation=${gtf} \\
        $cytobands \\
        $protein_domains \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        arriba: \$(arriba -h | grep 'Version:' 2>&1 |  sed 's/Version:\s//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.pdf
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        arriba: \$(arriba -h | grep 'Version:' 2>&1 |  sed 's/Version:\s//')
    END_VERSIONS
    """
}
