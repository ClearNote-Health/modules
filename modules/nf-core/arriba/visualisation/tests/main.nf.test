// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test arriba/visualisation
nextflow_process {

    name "Test Process ARRIBA_VISUALISATION"
    script "../main.nf"
    process "ARRIBA_VISUALISATION"

    tag "modules"
    tag "modules_nfcore"
    tag "arriba"
    tag "arriba/visualisation"

    test("homo_sapiens - fastq") {

        setup{
            run(STAR_GENOMEGENERATE) {
            script "../../../star/genomegenerate/main.nf"
            process {
            """
            fasta = [
                [ id:'test_fasta' ], // meta map
                [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
            fai = [
                [ id:'test_gtf' ], // meta map
                [ file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true) ]
            gtf = [
                [ id:'test_fasta_fai' ], // meta map
                [ file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true) ]
            """
            }
            }

            run(STAR_ALIGN) {
            script "../../../star/align/main.nf"
            process {
            """
            input = [ [ id:'test', single_end:false ], // meta map
                [   file(params.test_data['homo_sapiens']['illumina']['test_rnaseq_1_fastq_gz'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_rnaseq_2_fastq_gz'], checkIfExists: true) ]
            star_ignore_sjdbgtf = false
            seq_platform = 'illumina'
            seq_center = false
            STAR_GENOMEGENERATE.out.index
            gtf = [
                [ id:'test_fasta_fai' ], // meta map
                [ file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true) ]
            """
            }
            }
            run(ARRIBA_ARRIBA) {
            script "../../../arriba/arriba/main.nf"
            process {
            """
            STAR_ALIGN.out.bam
            fasta
            gtf
            """
            }
            }
            run(ARRIBA_DOWNLOAD) {
            script "../../../arriba/download/main.nf"
            process {
            """
            """
            }
            }

        }

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                STAR_GENOMEGENERATE ( fasta, gtf )
                STAR_ALIGN ( input, STAR_GENOMEGENERATE.out.index, gtf, star_ignore_sjdbgtf, seq_platform, seq_center )
                ARRIBA_ARRIBA ( STAR_ALIGN.out.bam, fasta, gtf, [[],[]], [[],[]], [[],[]], [[],[]], [[],[]])
                ARRIBA_DOWNLOAD ()
                ARRIBA_VISUALISATION ( STAR_ALIGN.out.bam, ARRIBA_ARRIBE.out.test.fusions.tsv, gtf, ARRIBA_DOWNLOAD.out.protein_domains_hg19_hs37d5_GRCh37_v2.4.0.gff3, ARRIBA_DOWNLOAD.out.cytobands_hg19_hs37d5_GRCh37_v2.4.0.tsv )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
