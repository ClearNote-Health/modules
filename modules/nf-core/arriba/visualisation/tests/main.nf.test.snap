{
    "homo_sapiens - fastq": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "pdf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "timestamp": "2024-05-30T15:40:44.701085"
    }
}